soapui/src/main/java/com/eviware/soapui/impl/support/HttpUtils.java

```diff
@@ -53,7 +53,7 @@ public static String completeUrlWithHttpIfProtocolIsNotHttpOrHttpsOrPropertyExpa
         if (StringUtils.isNullOrEmpty(endpoint)) {
             return endpoint;
         }
-        endpoint = endpoint.toLowerCase().trim();
+        endpoint = endpoint.trim();

```

Comment:
Hi Ole,
I made the changes and tests to cover the URL beginning with HTTP/HTTPS i.e.
`  
...
endpoint = endpoint.trim();
String lowerCaseEndpoint = endpoint.toLowerCase();
if (!lowerCaseEndpoint.startsWith("http://") && !lowerCaseEndpoint.startsWith("https://") && !endpoint.startsWith("$")) {
            return "http://" + endpoint;
}
...`
However, maybe as a second thought and counterpoint from a usability perspective, I would say that whilst this part of the code tries to help by automatically appending http://, it is actually potentially irritating in cases where the user is manually editing the protocol in the TestStep e.g. a bit like an over eager spellchecker that shoves in an unwanted correction! 
Basically that HTTP Request TestStep is a gem mainly because it is (almost) free format and lacks the structural restrictions on the REST Request TestStep, most people just want full control over the endpoint URL. So, what do you think about me instead refactoring the code to just trim the URL i.e. just

`if (StringUtils.isNullOrEmpty(endpoint)) {
            return endpoint;
  }
  endpoint = endpoint.trim();`

Let me know your thoughts.

Thanks for your time,
Rupert